[bug] [JOSS] MacOS Installation Issues

**Description**
Part of https://github.com/openjournals/joss-reviews/issues/3298.
The `readme` instructions fail:

```bash
cargo run --release examples/boron_nitride.toml
```

Causes:
<details>
<summary> Linking Error </summary>


```bash
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "-L" "/Users/rohitgoswami/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/rohitgoswami/Git/Github/JOSS/RustBCA/target/release/deps/libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o" "-o" "/Users/rohitgoswami/Git/Github/JOSS/RustBCA/target/release/deps/liblibRustBCA.dylib" "-Wl,-exported_symbols_list,/var/folders/jg/hl79b7sj291_ztvd_26cm4g40000gn/T/rustchFZYoG/list" "-Wl,-dead_strip" "-dynamiclib" "-Wl,-dylib" "-nodefaultlibs" "-L" "/Users/rohitgoswami/Git/Github/JOSS/RustBCA/target/release/deps" "-L" "/Users/rohitgoswami/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/rohitgoswami/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-10db70d883838cbc.rlib" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm"
  = note: Undefined symbols for architecture x86_64:
            "_PyErr_GivenExceptionMatches", referenced from:
                pyo3::types::module::PyModule::index::h64bd52c37ba1c695 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyExc_AttributeError", referenced from:
                pyo3::types::module::PyModule::index::h64bd52c37ba1c695 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyModule_GetName", referenced from:
                pyo3::types::function::PyCFunction::internal_new::h39b820cb301451f6 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_Py_InitializeEx", referenced from:
                parking_lot::once::Once::call_once_slow::hbb2eb33ab469899c in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyUnicodeDecodeError_Create", referenced from:
                pyo3::types::function::PyCFunction::internal_new::h39b820cb301451f6 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyGILState_Ensure", referenced from:
                pyo3::gil::GILGuard::acquire::h313e01bd2fd7501d in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyCFunction_NewEx", referenced from:
                pyo3::types::function::PyCFunction::internal_new::h39b820cb301451f6 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyUnicode_AsEncodedString", referenced from:
                pyo3::types::string::PyString::to_string_lossy::hc1a1e053e364b1e8 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyErr_Restore", referenced from:
                _PyInit_pybca in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                libRustBCA::__pyo3_raw_simple_bca_py::h19552f07e31f48ee in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                libRustBCA::__pyo3_raw_simple_bca_list_py::hcb93c0b96c415cae in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::err::PyErr::fetch::h782cc935f4026642 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyBytes_AsString", referenced from:
                pyo3::types::string::PyString::to_string_lossy::hc1a1e053e364b1e8 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyErr_PrintEx", referenced from:
                pyo3::err::PyErr::fetch::h782cc935f4026642 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyFloat_FromDouble", referenced from:
                pyo3::callback::convert::h3f1998a4c0d3c6fd in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyObject_Repr", referenced from:
                _$LT$$RF$T$u20$as$u20$core..fmt..Debug$GT$::fmt::h5d19767f6f043980 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                _$LT$pyo3..types..typeobject..PyType$u20$as$u20$core..fmt..Debug$GT$::fmt::hc903da73f1836966 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyTuple_Size", referenced from:
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyObject_GetAttr", referenced from:
                _$LT$$RF$T$u20$as$u20$core..fmt..Display$GT$::fmt::hc5f4df03e08fa065 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::index::h64bd52c37ba1c695 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::add_function::h1182a2f03290afec in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyExc_TypeError", referenced from:
                pyo3::panic::PanicException::new_err::h62c4791729ca2601 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::panic::PanicException::new_err::hef04c564e16a55d0 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::exceptions::PyTypeError::new_err::haafc783d740ff60d in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::err::PyErr::normalized::ha378f8f722ac6f8a in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::derive_utils::argument_extraction_error::hff03ef05b65c5bdf in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::function::PyCFunction::internal_new::h39b820cb301451f6 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyErr_NewException", referenced from:
                pyo3::err::PyErr::new_type::h0722a1986002f844 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PySequence_Size", referenced from:
                pyo3::types::any::PyAny::extract::h31195731a9c712b2 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyDict_Copy", referenced from:
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyDict_DelItem", referenced from:
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyGILState_Release", referenced from:
                core::ptr::drop_in_place$LT$pyo3..gil..EnsureGIL$GT$::h3e3ffab326cdbefa in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::panic::PanicException::new_err::h62c4791729ca2601 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::panic::PanicException::new_err::hef04c564e16a55d0 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::exceptions::PyTypeError::new_err::haafc783d740ff60d in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::err::PyErr::normalized::ha378f8f722ac6f8a in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                _$LT$pyo3..err..PyErr$u20$as$u20$core..fmt..Debug$GT$::fmt::h55ceb1ada545775f in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyDict_Size", referenced from:
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyObject_Str", referenced from:
                _$LT$$RF$T$u20$as$u20$core..fmt..Display$GT$::fmt::h615704b63f9d5624 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::derive_utils::argument_extraction_error::hff03ef05b65c5bdf in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_Py_IsInitialized", referenced from:
                parking_lot::once::Once::call_once_slow::hbb2eb33ab469899c in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyExc_SystemError", referenced from:
                pyo3::err::PyErr::normalized::ha378f8f722ac6f8a in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyEval_SaveThread", referenced from:
                parking_lot::once::Once::call_once_slow::hbb2eb33ab469899c in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyDict_Next", referenced from:
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyObject_GetIter", referenced from:
                pyo3::types::any::PyAny::extract::h31195731a9c712b2 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyErr_Occurred", referenced from:
                pyo3::types::any::PyAny::extract::h31195731a9c712b2 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                libRustBCA::__pyo3_raw_simple_bca_py::h19552f07e31f48ee in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                libRustBCA::__pyo3_raw_simple_bca_list_py::hcb93c0b96c415cae in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyEval_ThreadsInitialized", referenced from:
                parking_lot::once::Once::call_once_slow::hbb2eb33ab469899c in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyModule_Create2", referenced from:
                _PyInit_pybca in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyDict_GetItem", referenced from:
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyList_Append", referenced from:
                _PyInit_pybca in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::add_function::h1182a2f03290afec in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyErr_Fetch", referenced from:
                pyo3::err::PyErr::fetch::h782cc935f4026642 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyUnicode_AsUTF8AndSize", referenced from:
                pyo3::err::PyErr::fetch::h782cc935f4026642 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::string::PyString::to_string_lossy::hc1a1e053e364b1e8 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                _$LT$$RF$T$u20$as$u20$core..fmt..Display$GT$::fmt::hc5f4df03e08fa065 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::add_function::h1182a2f03290afec in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyObject_SetAttr", referenced from:
                _PyInit_pybca in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::index::h64bd52c37ba1c695 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::add_function::h1182a2f03290afec in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyBytes_Size", referenced from:
                pyo3::types::string::PyString::to_string_lossy::hc1a1e053e364b1e8 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyIter_Next", referenced from:
                pyo3::types::any::PyAny::extract::h31195731a9c712b2 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "__Py_Dealloc", referenced from:
                _PyInit_pybca in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                _$LT$pyo3..gil..GILPool$u20$as$u20$core..ops..drop..Drop$GT$::drop::hbee67efe74ede018 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::gil::ReferencePool::update_counts::heace5938242c0f82 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::gil::register_decref::hf721d52db974320d in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                _$LT$$RF$T$u20$as$u20$core..fmt..Display$GT$::fmt::hc5f4df03e08fa065 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::index::h64bd52c37ba1c695 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                ...
            "_PyErr_Print", referenced from:
                pyo3::err::panic_after_error::h9e14dc1fc2e89b1d in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyErr_NormalizeException", referenced from:
                pyo3::err::PyErr::normalized::ha378f8f722ac6f8a in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyTuple_New", referenced from:
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h88eba948f3580a0c in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::hab4dd6dd0eba8fd8 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyFloat_AsDouble", referenced from:
                pyo3::types::any::PyAny::extract::h31195731a9c712b2 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                libRustBCA::__pyo3_raw_simple_bca_py::h19552f07e31f48ee in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                libRustBCA::__pyo3_raw_simple_bca_list_py::hcb93c0b96c415cae in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyList_SetItem", referenced from:
                pyo3::callback::convert::h3f1998a4c0d3c6fd in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PySequence_Check", referenced from:
                pyo3::types::any::PyAny::extract::h31195731a9c712b2 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyTuple_GetItem", referenced from:
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyUnicode_FromStringAndSize", referenced from:
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h88eba948f3580a0c in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::hab4dd6dd0eba8fd8 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::hd0bf21ce6eccd88e in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                _PyInit_pybca in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h99e4ff115a1ab4f3 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::derive_utils::parse_fn_args::h6856090c433c0ad9 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h0059d36abecdead4 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                ...
            "_PyTuple_SetItem", referenced from:
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h88eba948f3580a0c in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::hab4dd6dd0eba8fd8 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyExc_BaseException", referenced from:
                pyo3::panic::PanicException::new_err::h62c4791729ca2601 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::panic::PanicException::new_err::hef04c564e16a55d0 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::err::PyErr::fetch::h782cc935f4026642 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
            "_PyList_New", referenced from:
                pyo3::callback::convert::h3f1998a4c0d3c6fd in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
                pyo3::types::module::PyModule::index::h64bd52c37ba1c695 in libRustBCA.libRustBCA.azzde3yo-cgu.0.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

error: could not compile `RustBCA`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```


</details>


**To Reproduce**
Please attach the relevant TOML input file and list the command line arguments used.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Error messages, output files, or figures**
Please include error messages, output files or figures that demonstrate the error if applicable.


**System (please complete the following information):**
OS: MacOS

![image](https://user-images.githubusercontent.com/4336207/127751845-d01a8077-fb63-4af9-9371-0fa63810c825.png)


**Additional context**
Add any other context about the problem here.

Works well enough on Linux systems, so perhaps the `readme` should be updated to reflect that.

--- a/.github/workflows/rustbca_compile_check.yml
+++ b/.github/workflows/rustbca_compile_check.yml
@@ -43,10 +43,10 @@ jobs:
         python3 -c "from libRustBCA.pybca import *; print(simple_bca_py)"
     - name: Test RustBCA
       run: |
-        cargo test --features cpr_rootfinder_netlib,hdf5_input,distributions,parry3d
+        sudo cargo test --features cpr_rootfinder_netlib,hdf5_input,distributions,parry3d
     - name: Run Examples
       run: |
-        cargo run --release 0D examples/boron_nitride_0D.toml
+        sudo cargo run --release 0D examples/boron_nitride_0D.toml
         ./target/release/RustBCA 0D examples/titanium_dioxide_0D.toml
         ./target/release/RustBCA 1D examples/layered_geometry_1D.toml
         cat 2000.0eV_0.0001deg_He_TiO2_Al_Sisummary.output
@@ -54,5 +54,5 @@ jobs:
         ./target/release/RustBCA examples/layered_geometry.toml
         cat 2000.0eV_0.0001deg_He_TiO2_Al_Sisummary.output
         ./target/release/RustBCA SPHERE examples/boron_nitride_sphere.toml
-        cargo run --release --features parry3d TRIMESH examples/tungsten_twist_trimesh.toml
+        sudo cargo run --release --features parry3d TRIMESH examples/tungsten_twist_trimesh.toml
         

--- a/.github/workflows/rustbca_compile_check.yml
+++ b/.github/workflows/rustbca_compile_check.yml
@@ -35,6 +35,11 @@ jobs:
     - name: Install HDF5 Libraries
       run: |
         sudo apt install libhdf5-dev
+    - name: test Python Bindings
+      run: |
+        suto python3 -m pip install setuptools_rust testresources
+        sudo python3 setup.py install
+        python3 -c "from libRustBCA.pybca import *; print(simple_bca_py)"
     - name: Test RustBCA
       run: |
         cargo test --features cpr_rootfinder_netlib,hdf5_input,distributions,parry3d
@@ -49,3 +54,4 @@ jobs:
         cat 2000.0eV_0.0001deg_He_TiO2_Al_Sisummary.output
         ./target/release/RustBCA SPHERE examples/boron_nitride_sphere.toml
         cargo run --release --features parry3d TRIMESH examples/tungsten_twist_trimesh.toml
+        

--- a/.github/workflows/rustbca_compile_check.yml
+++ b/.github/workflows/rustbca_compile_check.yml
@@ -47,12 +47,12 @@ jobs:
     - name: Run Examples
       run: |
         sudo cargo run --release 0D examples/boron_nitride_0D.toml
-        ./target/release/RustBCA 0D examples/titanium_dioxide_0D.toml
-        ./target/release/RustBCA 1D examples/layered_geometry_1D.toml
+        sudo ./target/release/RustBCA 0D examples/titanium_dioxide_0D.toml
+        sudo ./target/release/RustBCA 1D examples/layered_geometry_1D.toml
         cat 2000.0eV_0.0001deg_He_TiO2_Al_Sisummary.output
-        ./target/release/RustBCA examples/boron_nitride.toml
-        ./target/release/RustBCA examples/layered_geometry.toml
+        sudo ./target/release/RustBCA examples/boron_nitride.toml
+        sudo ./target/release/RustBCA examples/layered_geometry.toml
         cat 2000.0eV_0.0001deg_He_TiO2_Al_Sisummary.output
-        ./target/release/RustBCA SPHERE examples/boron_nitride_sphere.toml
+        sudo ./target/release/RustBCA SPHERE examples/boron_nitride_sphere.toml
         sudo cargo run --release --features parry3d TRIMESH examples/tungsten_twist_trimesh.toml
         

--- a/.github/workflows/rustbca_compile_check.yml
+++ b/.github/workflows/rustbca_compile_check.yml
@@ -21,6 +21,7 @@ jobs:
     - name: Install rust
       run: |
         curl --proto '=https' --tlsv1.2 -sSf -y https://sh.rustup.rs | sh
+        sudo apt-get install rustc
     - name: Install pip for Python-3
       run: |
         sudo apt-get install python3-pip

--- a/.github/workflows/rustbca_compile_check.yml
+++ b/.github/workflows/rustbca_compile_check.yml
@@ -37,7 +37,7 @@ jobs:
         sudo apt install libhdf5-dev
     - name: test Python Bindings
       run: |
-        suto python3 -m pip install setuptools_rust testresources
+        sudo python3 -m pip install setuptools_rust testresources
         sudo python3 setup.py install
         python3 -c "from libRustBCA.pybca import *; print(simple_bca_py)"
     - name: Test RustBCA

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,6 +32,7 @@ parry3d-f64 = {version = "0.2.0", optional = true}
 [dependencies.pyo3]
 version = "0.13.2"
 features = ["extension-module"]
+optional = true
 
 [dev-dependencies]
 float-cmp = "0.8.0"
@@ -51,3 +52,4 @@ distributions = ["ndarray"]
 no_list_output = []
 parry3d = ["parry3d-f64"]
 accelerated_ions = []
+python = ["pyo3"]

--- a/setup.py
+++ b/setup.py
@@ -5,5 +5,6 @@ setup(
     name="RustBCA",
     rust_extensions=[RustExtension("libRustBCA.pybca", binding=Binding.PyO3)],
     # rust extensions are not zip safe, just like C-extensions.
+    features=["python"]
     zip_safe=False,
 )

--- a/setup.py
+++ b/setup.py
@@ -3,8 +3,7 @@ from setuptools_rust import Binding, RustExtension
 
 setup(
     name="RustBCA",
-    rust_extensions=[RustExtension("libRustBCA.pybca", binding=Binding.PyO3)],
+    rust_extensions=[RustExtension("libRustBCA.pybca", binding=Binding.PyO3, features=["python"])],
     # rust extensions are not zip safe, just like C-extensions.
-    features=["python"],
     zip_safe=False,
 )

--- a/setup.py
+++ b/setup.py
@@ -5,6 +5,6 @@ setup(
     name="RustBCA",
     rust_extensions=[RustExtension("libRustBCA.pybca", binding=Binding.PyO3)],
     # rust extensions are not zip safe, just like C-extensions.
-    features=["python"]
+    features=["python"],
     zip_safe=False,
 )

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -39,7 +39,9 @@ use std::f64::consts::FRAC_2_SQRT_PI;
 use std::f64::consts::PI;
 use std::f64::consts::SQRT_2;
 
+#[cfg(feature = "python")]
 use pyo3::prelude::*;
+#[cfg(feature = "python")]
 use pyo3::wrap_pyfunction;
 
 //Load internal modules
@@ -70,6 +72,7 @@ pub use crate::sphere::{Sphere, SphereInput, InputSphere};
 #[cfg(feature = "parry3d")]
 pub use crate::parry::{ParryBall, ParryBallInput, InputParryBall, ParryTriMesh, ParryTriMeshInput, InputParryTriMesh};
 
+#[cfg(feature = "python")]
 #[pymodule]
 pub fn pybca(py: Python, m: &PyModule) -> PyResult<()> {
     m.add_function(wrap_pyfunction!(simple_bca_py, m)?)?;
@@ -467,11 +470,13 @@ pub extern "C" fn simple_bca_c(x: f64, y: f64, z: f64, ux: f64, uy: f64, uz: f64
     }
 }
 
+#[cfg(feature = "python")]
 #[pyfunction]
 pub fn simple_bca_py(x: f64, y: f64, z: f64, ux: f64, uy: f64, uz: f64, E1: f64, Z1: f64, m1: f64, Ec1: f64, Es1: f64, Z2: f64, m2: f64, Ec2: f64, Es2: f64, n2: f64, Eb2: f64) -> Vec<[f64; 9]> {
     simple_bca(x, y, z, ux, uy, uz, E1, Z1, m1, Ec1, Es1, Z2, m2, Ec2, Es2, n2, Eb2)
 }
 
+#[cfg(feature = "python")]
 #[pyfunction]
 pub fn simple_bca_list_py(energies: Vec<f64>, usx: Vec<f64>, usy: Vec<f64>, usz: Vec<f64>, Z1: f64, m1: f64, Ec1: f64, Es1: f64, Z2: f64, m2: f64, Ec2: f64, Es2: f64, n2: f64, Eb2: f64) -> Vec<[f64; 9]> {
 

